### DISCUSSION TITLE
How to support multiple storage locations with the same postgres database

### DISCUSSION THREAD

**Discussion by sepatel** [juicedata/juicefs#4679] [Q&A]
I don't really know how to ask this and it is super easy to do with redis. I have 2 different S3 buckets with different access/secret keys that I need to mount at different locations. However, we have an initiative to move them to postgres instead of redis for the metadata. These are new mounts and we will manually copy the actual content over.

Doing this in redis is easy because I can do something like
```
juicefs format \
  --storage s3 \
  --bucket https://example1.s3.us-east-1.amazonaws.com \
  --access-key ABC \
  --secret-key ABC \
  --compress zstd \
  --trash-days 0 \
  "redis://me:pass@redis1:6379,redis2:6379/1" \
  artifacts

juicefs format \
  --storage s3 \
  --bucket https://example2.s3.us-east-1.amazonaws.com \
  --access-key XYZ \
  --secret-key XYZ \
  --compress zstd \
  --trash-days 0 \
  "redis://me:pass@redis1:6379,redis2:6379/2" \
  artifacts
```

and the `/1` and `/2` of redis allows for multiple logical allocations. However, if I try doing the same thing but instead of redis doing something like
```
"postgres://postgres:password@postgres:5432/myapp"
```

it works for the first but errors for the second. I don't have the option of creating a new database schema for every new setup. Is there a way to pass a flag or something to provide it with a logical id like how it works for redis. What is the way to support multiple storages using the same database? FYI, I have over 300 of these to do.

**Comment by sinofool** [comment]
what about using not the same database, but on the same postgres server?
`"postgres://postgres:password@postgres:5432/mysecondapp"`

**Reply by sepatel** [reply]
I am doing that right now, have over a 100 schemas defined as each one has to be it's own database instance. But from a security perspective, operations tends to not like for software to be creating schemas automatically from scratch and having 100's eventually 1000's of schemas makes it difficult to do other types of things like backups/restores and such as well. It becomes a bit of an operational nightmare so to speak.

Edit: also from an application perspective, I'm unsure what all is happening exactly but I told that the number of connections is going to be higher from a connection pool perspective as well creating a lot of uneeeded resource usages and increased costs.

Edit 2: I'm reminded that because of the security violations caused by this, they actually had to spin us off into a completely different system and network as the application needed to have god-like access to the database which is in violation of all best practices for a software to do that.
